lib/SILGen/SILGenApply.cpp

```diff
@@ -5006,6 +5006,44 @@ RValue SILGenFunction::emitApplyMethod(SILLocation loc, ConcreteDeclRef declRef,
   return emission.apply(C);
 }
 
+RValue SILGenFunction::emitApplyOfPropertyWrapperBackingInitializer(
+    SILLocation loc,
+    VarDecl *var,
+    RValue &&originalValue,
+    SGFContext C) {
+  SILDeclRef constant(var, SILDeclRef::Kind::PropertyWrapperBackingInitializer);
+
+  SubstitutionMap subs;
+  auto varDC = var->getInnermostDeclContext();
+  if (auto genericSig = varDC->getGenericSignatureOfContext()) {
+    subs = SubstitutionMap::get(
+        genericSig,
+        [&](SubstitutableType *type) {
+          if (auto gp = type->getAs<GenericTypeParamType>()) {
+            return F.mapTypeIntoContext(gp);
+          }
+
+          return Type(type);
+        },
+        LookUpConformanceInModule(varDC->getParentModule()));
+  }
+
+  FormalEvaluationScope writebackScope(*this);
```

Comment:
You may want an argument scope here instead of a FormalEvaluationScope: @slavapestov your thoughts?

Reply:
Or actually, that may already be in CallEmission. I forgot.

Reply:
It looks like `CallEmission::apply()` calls `CallEmission::applyFirstLevelCallee()`, which calls `CallEmission::applyNormalCall()`, which creates an `ArgumentScope` before it calls `CallEmission::emitArgumentsForNormalApply()`.